src/components/MultiHopTrade/components/LimitOrder/components/CancelLimtOrder.tsx

```diff
@@ -0,0 +1,222 @@
+import {
+  Button,
+  Card,
+  Flex,
+  Heading,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Progress,
+  Stack,
+} from '@chakra-ui/react'
+import { SwapperName } from '@shapeshiftoss/swapper'
+import type { CowSwapError } from '@shapeshiftoss/types'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { bn, bnOrZero, fromBaseUnit } from '@shapeshiftoss/utils'
+import { formatDistanceToNow } from 'date-fns'
+import { useCallback, useEffect, useMemo } from 'react'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIconWithBadge } from 'components/AssetIconWithBadge'
+import { Row } from 'components/Row/Row'
+import { RawText, Text } from 'components/Text'
+import { TransactionTypeIcon } from 'components/TransactionHistory/TransactionTypeIcon'
+import { useActions } from 'hooks/useActions'
+import { useErrorHandler } from 'hooks/useErrorToast/useErrorToast'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { useCancelLimitOrdersMutation } from 'state/apis/limit-orders/limitOrderApi'
+import { limitOrderSlice } from 'state/slices/limitOrderSlice/limitOrderSlice'
+import { selectOrderToCancel } from 'state/slices/limitOrderSlice/selectors'
+import { selectAssetById, selectFeeAssetById } from 'state/slices/selectors'
+import { useAppSelector, useSelectorWithArgs } from 'state/store'
+
+import { SwapperIcon } from '../../TradeInput/components/SwapperIcon/SwapperIcon'
+
+const cardBorderRadius = { base: '2xl' }
+
+export const CancelLimitOrder = () => {
+  const wallet = useWallet().state.wallet
+  const { setOrderToCancel } = useActions(limitOrderSlice.actions)
+  const { showErrorToast } = useErrorHandler()
+
+  const orderToCancel = useAppSelector(selectOrderToCancel)
+
+  const [cancelLimitOrders, { data: wasCancellationSuccessful, error, isLoading, reset }] =
+    useCancelLimitOrdersMutation()
+
+  useEffect(() => {
+    if (!error) return
+
+    const description = (error as CowSwapError).description ?? 'Unknown Error'
+
+    // TODO: Actually render a translated error description.
```

Comment:
q: are we ever going to? i.e do we feel like we can surface all CoW errors (including unknown unknowns since the API is "sometimes undocumented" to put it that way) as nice translations, or can we just assume order cancelation is straightforward enough that in case of any errors, displaying response error description as-is is enough?

Reply:
Agree, I'll remove the TODO and if we ever want it we can add them.

Reply:
2151427d82837a2c91febcaed514dcce0fdc662d